### PR TITLE
Fix delete event before values is null at pipeline DataRecord

### PR DESCRIPTION
**Pull request by azexcy**
Changes proposed in this pull request:
  - Fix delete event before values is null at pipeline DataRecord

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataSourceImporter.java [hunk 1]
```diff
@@ -221,7 +221,7 @@ private void executeUpdate(final Connection connection, final DataRecord record)
             }
             for (int i = 0; i < conditionColumns.size(); i++) {
                 Column keyColumn = conditionColumns.get(i);
-                preparedStatement.setObject(updatedColumns.size() + i + 1, keyColumn.isUniqueKey() && keyColumn.isUpdated() ? keyColumn.getOldValue() : keyColumn.getValue());
+                preparedStatement.setObject(updatedColumns.size() + i + 1, keyColumn.getOldValue());
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataSourceImporter.java:224]: `keyColumn.getOldValue()` might be `null`?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java [hunk 2]
```diff
@@ -119,21 +119,22 @@ private void mergeUpdate(final DataRecord dataRecord, final Map<DataRecord.Key,
     }
     
     private void mergeDelete(final DataRecord dataRecord, final Map<DataRecord.Key, DataRecord> dataRecords) {
-        DataRecord beforeDataRecord = dataRecords.get(dataRecord.getKey());
+        DataRecord beforeDataRecord = dataRecords.get(dataRecord.getOldKey());
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java:122]: 1, `dataRecord.getOldKey()` might be `null`?

2, Could `dataRecord.getOldKey()` method name be updated?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java [hunk 3]
```diff
@@ -119,21 +119,22 @@ private void mergeUpdate(final DataRecord dataRecord, final Map<DataRecord.Key,
     }
     
     private void mergeDelete(final DataRecord dataRecord, final Map<DataRecord.Key, DataRecord> dataRecords) {
-        DataRecord beforeDataRecord = dataRecords.get(dataRecord.getKey());
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java:122]: Could we remove `mergeDelete` and `merge` methods etc

File: kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/ingest/record/Column.java [hunk 4]
```diff
@@ -40,10 +40,6 @@ public final class Column {
     
     private final boolean uniqueKey;
     
-    public Column(final String name, final Object value, final boolean updated, final boolean uniqueKey) {
-        this(name, null, value, updated, uniqueKey);
-    }
```
Comment (sandynz) [inline kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/ingest/record/Column.java:45]: Could we keep this constructor? To reduce unrelated changes

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java (hunk 3 above)
Comment (azexcy) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java:122]: I'll think about it later

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java (hunk 2 above)
Comment (azexcy) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java:122]: Delete record before have value, after is nul, the getOldKey() will get the before uniquke value, is't not null

**Review by sandynz — APPROVED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java (hunk 2 above)
Comment (sandynz) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataRecordMerger.java:122]: OK
